src/main/java/gift/main/admin/AdminController.java

```diff
@@ -0,0 +1,85 @@
+import gift.main.entity.Product;
+import gift.main.handler.ProductTransformer;
+import gift.main.repository.ProductDao;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+
+@Controller
+@RequestMapping("/admin")
+public class AdminController {
+    private final ProductDao productDao;
+    private final AtomicLong idGenerator = new AtomicLong();
+
+    public AdminController(ProductDao productDao) {
+        this.productDao = productDao;
+        productDao.createProductTable();
+    }
+
+
+    @GetMapping({"","/"})
+    public String adminPage(Model model) {
+        model.addAttribute("products", productDao.selectProductAll());
+        return "product";
+    }
+
+
+    @GetMapping("/product")
+    public String findProduct(@RequestParam(value = "id") long id,Model model) {
+        if (productDao.selectProduct(id)==null) {
+            model.addAttribute("messages","해당아이디값은 없습니다.");
+            model.addAttribute("products",productDao.selectProductAll());
+            return "product";
+        }
+        model.addAttribute("seletProduct", productDao.selectProduct(id));
+        model.addAttribute("products", productDao.selectProductAll());
+        return "product";
+
+    }
+
+    @PostMapping("/product")
+    public String addProduct(@ModelAttribute ProductRequest productRequest,Model model) {
+        long id = idGenerator.incrementAndGet();
+        productDao.insertProduct(ProductTransformer.convertToProduct(id, productRequest));
+
+        model.addAttribute("products",  productDao.selectProductAll());
+        return "product";
+    }
+
+    @PutMapping(value = "/product")
+    public String updateProduct(@RequestParam(value = "id") long id, @ModelAttribute ProductRequest productRequest, Model model){
+        productDao.updateProduct(id, ProductTransformer.convertToProduct(id, productRequest));
+        model.addAttribute("products", productDao.selectProductAll());
+        return "product";
+    }
+
+    @DeleteMapping("/product")
+    public String deleteProduct(@RequestParam(value = "id") long id, Model model) {
+        productDao.deleteProduct(id);
+        model.addAttribute("products", productDao.selectProductAll());
+        return "product";
+    }
+
+//    private List<Product> getProduct() {
+//        List<Product> productList = new ArrayList<>();
+//        for (Map.Entry<Long, Product> productEntry : productsRepository.entrySet()) {
+//            productList.add(productEntry.getValue());
+//        }
+//
+//        return productList;
+//    }
+
+    @GetMapping("/product/edit")
+    public String editPage(@RequestParam(value = "id") long id, @ModelAttribute ProductRequest productRequest,Model model) {
+        Product product = ProductTransformer.convertToProduct(id, productRequest);
+        model.addAttribute("product", product);
+        return "edit";

```

Comment:
```suggestion
    @GetMapping("/product/edit")
    public String editPage(
        @RequestParam(value = "id") long id,
        @ModelAttribute ProductRequest productRequest,
        Model model
    ) {
        Product product = ProductTransformer.convertToProduct(id, productRequest);
        model.addAttribute("product", product);
        return "edit";
    }
```

변수 중간에 띄어쓰기기 있는 곳도 있고, 없는 곳도 있어 Refomat 등으로 전체적인 코드 컨벤션을 맞춰져도 좋을 것 같습니다. 

저는 파라미터가 2개 이상이되면 한줄에 모두 쓰지 않고 한줄씩 띄어쓰는데 이 부분은 취향이라서 반영하지 않으셔도 좋습니다😎

Reply:
호호 저는 수정하고 싶습니다. 개인적인 이유지만 멋진 코드가 좋으니까요..!! 변수 컨벤션을 미리 정하는 것이 좋을 것 같습니다! 코드는 모두의 코드니까요..!!